server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestClusterStatsAction.java

```diff
@@ -28,6 +30,8 @@
 public class RestClusterStatsAction extends BaseRestHandler {
 
     private static final Set<String> SUPPORTED_CAPABILITIES = Set.of("human-readable-total-docs-size");
+    private static final Set<String> SUPPORTED_CAPABILITIES_CCS_STATS = Sets.union(SUPPORTED_CAPABILITIES, Set.of("ccs-stats"));
+    public static final FeatureFlag CCS_TELEMETRY_FEATURE_FLAG = new FeatureFlag("ccs_telemetry");
```

Comment:
I don't think we need a feature flag for this, let's just enable it always.

Reply:
The underlying data is not available though if the flag is not enabled, so it may be hard to test this way because we couldn't use capabilities to see what result is expected.

Reply:
We certainly need more work on the BwC situation in this area, dealing with remote clusters that don't know how to expose their remote-cluster-only stats. But a feature flag won't help with that, in fact I think it makes it harder.

Reply:
I assume the flag is temporary and we'll drop it soon, but while it's there I understand tests will be running both with and without the flag, including possibly YAML tests? And given that, we'd need some way to see whether the test result is expected to include the css telemetry? Is there any other way to see that?

Reply:
You mean like the situation when the local cluster is say 8.16 and the remote ones are pre-8.16 so they can't interpret `TransportRemoteClusterStatsAction` request? But in that case, wouldn't they just return an appropriate error response and the local cluster will deal with that? Or is there some other scenario that we aren't handling properly?

Reply:
I think it'll return an error in production yeah but IIRC we consider this a bug and assert that it doesn't happen in tests. We shouldn't be sending incomprehensible requests to nodes, we should be checking the `TransportVersion` first and bailing out early instead.

Reply:
OK I see what you mean. We don't have currently tests for that anywhere since the remote functionality requires request parameter (include_remotes) and none of the bwc tests does that - should we add it? In any case though, when I am initiating the remote request, I may not even have the connection to the remote cluster yet - how would I know the transport version? I don't see any API in `RemoteClusterClient` that allows to condition requests on the remote being of particular version?

Reply:
Yes the API isn't very BwC-sensitive here is it? Hmm. I think the best we can do right now is to check the transport version in `RemoteClusterStatsRequest#writeTo` and throw a `UnsupportedOperationException`.

Reply:
Oh and FWIW I think we should default `include_remotes` to `true`.

Reply:
The product requirement is to not to produce the remotes part by default, because it's expensive and is only needed once a day for Kibana to collect it. The data there is a copy of the data that is available on remotes anyway, so it wouldn't make much sense for a human to use this to look at it. It's more so that Kibana collector could have this data.

Reply:
> to check the transport version in RemoteClusterStatsRequest#writeTo and throw a UnsupportedOperationException.

Wait, in `writeTo` transport version is remote's transport? I didn't know that. If so, then yes, we can add a check there. Not sure how to test it...

Reply:
It's the negotiated transport version of the connection between the two nodes, which means the min of their transport versions. So yeah this'll tell you if the remote doesn't understand what on earth we're talking about.

Reply:
> The product requirement is to not to produce the remotes part by default, because it's expensive and is only needed once a day for Kibana to collect it.

I see, ok.

> The data there is a copy of the data that is available on remotes anyway, so it wouldn't make much sense for a human to use this to look at it.

I'm not sure that's 100% true. I think this could well be valuable to capture in a diagnostics bundle for instance, we do get support cases where knowing some details about the remote clusters would be helpful. Still, ok to not collect by default, but it'd be great to add this option in https://github.com/elastic/support-diagnostics/blob/main/src/main/resources/elastic-rest.yml as a follow-up.

Reply:
Ok I added the version check. Are there any docs on elastic-rest.yaml? That's the first time I encounter it.

Reply:
Just the comment at the top I think.

Reply:
I have added a followup task for updating support-diagnostics, once the main code is merged.